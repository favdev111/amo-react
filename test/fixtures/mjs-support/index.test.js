const testSetup = require('../__shared__/test-setup');

const puppeteer = require('puppeteer');

test('can use mjs library in development', async () => {
  const { port, done } = await testSetup.scripts.start();

  const browser = await puppeteer.launch({ headless: true });
  try {
    const page = await browser.newPage();
    await page.goto(`http://localhost:${port}/`);
    await page.waitForSelector('.Pokemon-Name-Data', { timeout: 0 });
    const output = await page.evaluate(() => {
      return Array.from(
        document.getElementsByClassName('Pokemon-Name-Data')
      ).pop().innerHTML;
    });
    expect(output).toMatchSnapshot();
  } finally {
    browser.close();
    done();
  }
});
test('can use mjs library in production', async () => {
  await testSetup.scripts.build();
  const { port, done } = await testSetup.scripts.serve();

  const browser = await puppeteer.launch({ headless: true });
  try {
    const page = await browser.newPage();
    await page.goto(`http://localhost:${port}/`);
    await page.waitForSelector('.Pokemon-Name-Data', { timeout: 0 });
    const output = await page.evaluate(() => {
      return Array.from(
        document.getElementsByClassName('Pokemon-Name-Data')
      ).pop().innerHTML;
    });
    expect(output).toMatchSnapshot();
  } finally {
    browser.close();
    done();
  }
});
