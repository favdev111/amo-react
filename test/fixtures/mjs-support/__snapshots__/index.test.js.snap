// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can use mjs library in development 1`] = `"Pikachu"`;

exports[`can use mjs library in production 1`] = `"Pikachu"`;
